activesupport/lib/active_support/callbacks.rb

```diff
@@ -500,7 +510,7 @@ def self.build(filter, callback)
           when ::Proc
             if filter.arity > 1
               InstanceExec2.new(filter)
-            elsif filter.arity > 0
+            elsif filter.arity > 0 || filter.arity < -1
```

Comment:
Even `-1` could take multiple arguments.

So the better condition would be:

- if arity = 0
- elseif arity = 1
- else

Reply:
I agree! The logic for greater than was a bit confusing, so I changed it to use equal, which made it much clearer. Thank you for pointing that out!

I was initially trying to avoid changing too much code, but I think this change make sense